### DISCUSSION TITLE
Install fails at CA step

### DISCUSSION THREAD

**Discussion by SpaceCoyote300** [Security-Onion-Solutions/securityonion#2657] [Unsupported Versions]
(ISO install, v2.3.21) My install is failing at the CA install step salt_checkin with the below throw. It looks like the salt key is being created but not accepted. I see the key has not been accepted if I run a salt-key -L debug, and I can accept the key without error, but re-running the install after that doesn't help because it flushes the salt state. I tried adding a line to accept all keys in the salt_checkin function, but so far no luck getting past this step. My install is entirely offline, and I can't connect my environment to the internet. Here is the error from sosetup.log:
`ERROR - The Salt Master has cached the public key for this node`
Is there a way I can skip this step in the installer after I manually accept the minion keys?
Also, how do I specify an automation/answer file when calling so-setup?
Thank you!

**Comment by m0duspwnens** [comment]
Which type of node install is this, manager, sensor, standalone, etc? Can you post the setup log?

Thanks.

**Reply by SpaceCoyote300** [reply]
Standalone. The environment I'm installing in is secluded, so I need to copy everything by hand. I'll try and copy as much of the log as seems relevant, following the error.
Edit: Okay, found a way to get screengrabs out
Edit2: It looks like intel.dat is missing and might kick off the start of the failure. I confirmed that the file says that it copied successfully higher in the log from zeek/policy/intel/intel.dat
Edit3: you'll see that it says that hostname_standalone doesn't match any unaccepted keys, but when I run salt-key -L I see hostname_standalone is there, and is unaccepted
Edit4: Tried changing 'salt-key -ya "$MINION_ID"' to 'salt-key -yA' and the error changed to 'they key glob * does not match any unaccepted keys', and the key still isn't accepted after the installer completes, so I tried and add another approve command near the STR 23 where the error is being thrown with the same 'key glob *' error above.

![rpviewer (2)](https://user-images.githubusercontent.com/77597149/105185397-779e7b00-5afe-11eb-8c43-9b99b32293e1.png)
![rpviewer (1)](https://user-images.githubusercontent.com/77597149/105185398-779e7b00-5afe-11eb-9c55-7211202fb589.png)

**Reply by ScurvySour** [reply]
Did you figure out any solution? I'm having a similar problem with what seems like a missing CA certificate when doing an air-gapped installation.

**Reply by SpaceCoyote300** [reply]
No fix yet. I'm reinstalling from scratch now. Nothing above ended up helping at all. 

Edit: Wondering if it has anything to do with the incredible amount of time it takes to run the post-installation scripts. The step 'Running... /bin/sh /tmp/ks-script-S7LzM9' took many hours (the last six alphanumbs were different last time, pretty sure those are generated), and last run-through I had to run that script myself from the shell window to get it to complete, it never completed on it's own even being allowed to run overnight.

**Reply by m0duspwnens** [reply]
@SpaceCoyote300 
 Sorry for the delay. I think the issue is that the Salt minion key is being generated, then the script is moving on to accept the key, but the key hasn't been fully generated yet. We have a change in the current dev branch the will loop and recheck for the key, but for now, I think we can just add a sleep to your local copy of the setup script before running it.

 Lets first revert any changes made, or pull a fresh copy of the 2.3.21 branch. Then modify so-setup and add a sleep between when the Salt minion key is generated and the Salt master accepting it. After [line 612](https://github.com/Security-Onion-Solutions/securityonion/blob/master/setup/so-setup#L612) in so-setup, add `sleep 120`, save and close the file. Run the install. If it still says that the key isn't available to be accepted, we may need to increase the sleep time.

 What are the hardware specs of this machine?

**Reply by SpaceCoyote300** [reply]
Dell PowerEdge R630, 2xXeon E5-2630 v3 2.4GHz (16 cores), 32GB DDR-4 (1866MHz)

**Reply by SpaceCoyote300** [reply]
Okay, setup got passed the CA step (Edit: clarifying: inserting sleep 120 at line 612 of so-setup worked), but is now dieing at 26% trying to download SO-ACNG. This is an airgap install, so I'm not sure why it's trying to download anything...
`Error response from daemon: Get https://quay.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
...
curl: (6) Could not resolve host: sigs.securityonion.net; Unknown error
Unable to pull signature file for so-acng:2.3.21`

Edit: a word

**Reply by ScurvySour** [reply]
I did not manage to get any further implementing the sleep timer. However, I tried to install 2.3.10 instead and that one also fails at 26% while trying to download SO-ACNG.

**Reply by SpaceCoyote300** [reply]
So far I've not found any way around this by tweaking the install code, anyone else have any luck?

**Reply by ScurvySour** [reply]
in a VM i was able to get through the install by disabling the network connection. This has to be done after the network is setup by the script, but before the rest of the process, for instance by going to another terminal and using nmtui. 

I think it relates to an external get request that times out instead of being denied.
